src/client.ts

```diff
@@ -145,9 +147,20 @@ export class JSONRPCClient<ClientParams = void>
     if (response.result !== undefined && !response.error) {
       return response.result;
     } else if (response.result === undefined && response.error) {
-      return Promise.reject(new Error(response.error.message));
+      return Promise.reject(
+        new JSONRPCErrorException(
+          response.error.message,
+          response.error.code,
+          response.error.data
+        )
+      );
     } else {
-      return Promise.reject(new Error("An unexpected error occurred"));
+      return Promise.reject(
+        new JSONRPCErrorException(
+          "An unexpected error occurred",
+          DefaultErrorCode
+        )
+      );
```

Comment:
It shouldn't be an `JSONRPCErrorException`. It's an regular error:
```suggestion
      return Promise.reject(new Error("An unexpected error occurred"));
```